common/addons/config-ore/src/main/java/com/dfsek/terra/addons/ore/ores/VanillaScatteredOre.java

```diff
@@ -0,0 +1,57 @@
+package com.dfsek.terra.addons.ore.ores;
+
+import com.dfsek.terra.addons.ore.utils.VanillaOreUtils;
+import com.dfsek.terra.api.block.BlockType;
+import com.dfsek.terra.api.block.state.BlockState;
+import com.dfsek.terra.api.structure.Structure;
+import com.dfsek.terra.api.util.Rotation;
+import com.dfsek.terra.api.util.collection.MaterialSet;
+import com.dfsek.terra.api.util.vector.Vector3;
+import com.dfsek.terra.api.util.vector.Vector3Int;
+import com.dfsek.terra.api.world.WritableWorld;
+
+import java.util.Map;
+import java.util.Random;
+
+import static com.dfsek.terra.addons.ore.utils.VanillaOreUtils.shouldPlace;
+
+
+public class VanillaScatteredOre extends VanillaOre {
+    protected final int spread;
+    public VanillaScatteredOre(BlockState material, double size, MaterialSet replaceable, boolean applyGravity, double exposed,
+                               Map<BlockType, BlockState> materials, int spread) {
+        super(material, size, replaceable, applyGravity, exposed, materials);
+        
+        this.spread = spread;
+    }
+    
+    @Override
+    public boolean generate(Vector3Int location, WritableWorld world, Random random, Rotation rotation) {
+        int i = random.nextInt((int) (size + 1));
+        Vector3Int.Mutable mutable = Vector3Int.zero().mutable();
+        
+        for(int j = 0; j < i; ++j) {
+            this.setPos(mutable, random, location, Math.min(j, spread));
+            BlockType block = world.getBlockState(mutable).getBlockType();
+                if (shouldPlace(getReplaceable(), block, exposed, random, world, mutable.getX(), mutable.getY(), mutable.getZ())) {
+                    world.setBlockState(mutable, getMaterial(block), isApplyGravity());
+                    break;
```

Comment:
This break statement shouldn't be here, otherwise only one ore block will ever be placed. MC source uses the break statement to stop when the first configured block state predicate matches, not to break out of iterating over each random position.